### ISSUE TITLE
Secure Headers' config.hsts overriden by Rails' config.force_ssl

### ISSUE THREAD

**Issue by encoder** [twitter/secureheaders#237]
I have added Secure Headers to my webapp (for CSP), and so far it has worked great.

Now I am trying to implement HSTS preloading, and added `config.hsts = "max-age=#{20.years.to_i}; includeSubdomains; preload"`, but noticed that Rails still sends a different "Strict-Transport-Security" header, which misses the "preload" keyword.

Upon investigating, it appears that because my Rails environment includes `config.force_ssl = true` (in order to redirect http->https, mark cookies as secure, etc.), Rails overwrites the "Strict-Transport-Security" header. The relevant code (for Rails 4.2) is here:

https://github.com/rails/rails/blob/4-2-stable/actionpack/lib/action_dispatch/middleware/ssl.rb#L48

Is it possible for Secure Headers to take priority over that other middleware?

**Comment by jacobbednarz** [comment]
Looks like https://github.com/twitter/secureheaders/blob/master/lib/secure_headers/railtie.rb#L13 is setting it to be the first in line but should this be after the `ActionDispatch::SSL` to ensure it doesn't get overridden?

@encoder to get around your issue you could always look to pop something like this in your `config/application.rb`.

```
config.middleware.insert_after(ActionDispatch::SSL, SecureHeaders::Middleware)
```

(Syntax may be slightly off so check it before just dropping it in)

**Comment by jacobbednarz** [comment]
To maintain backwards compatibility with Rails 3 and non-`ActionDispatch::SSL` applications we might need to the middleware to look something like this:

```rb
initializer "secure_headers.middleware" do
  if defined?(ActionDispatch::SSL)
    Rails.application.config.middleware.insert_after ActionDispatch::SSL, SecureHeaders::Middleware
  else
    Rails.application.config.middleware.insert_before 0, SecureHeaders::Middleware
  end
end
```

**Comment by oreoshake** [comment]
That looks great @jacobbednarz.

**Comment by jacobbednarz** [comment]
Pull request to follow shortly.

**Comment by asimenos** [comment]
See discussion in #241; looks fixed in versions >= 3.1.0

**Comment by oreoshake** [comment]
🎊
